### PR TITLE
gotestsum: Init at 0.3.5

### PR DESCRIPTION
**Pull request by endocrimes**
###### Motivation for this change

Introduce a nix package for `gotestsum`, a golang test runner that provides both better human friendly output and also machine readable junit xml without having to run tests multiple times. It's fairly new but used heavily by go projects that I interact with.

I'm adding it here rather than keeping it in my personal package repository as it may provide value to others.

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/gotestsum/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  version = "0.3.5";
+
+  src = fetchFromGitHub {
+    owner = "gotestyourself";
+    repo = "gotestsum";
+    rev = "v${version}";
+    sha256 = "1d4sbvk9wqzl3g3da8inqdkvd43rkwvmq969jlgl1k1agv5xjxqv";
+  };
+
+  modSha256 = "1q1rx3m25zw1xx3srfpaqmazgw987hdkpd9lhxcg2fg5qfz97psd";
```
[the remark this answers is not in the document]
Comment (endocrimes) [reply pkgs/development/tools/gotestsum/default.nix]: Ah - I had an env issue that meant I was pointing to the wrong nixpkgs for the build, and got the wrong go version :sob: - Fixed, thanks!

File: pkgs/development/tools/gotestsum/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{ stdenv, fetchFromGitHub, buildGoModule }:
+
+buildGoModule rec {
+  name = "gotestsum-${version}";
```
[the remark this answers is not in the document]
Comment (endocrimes) [reply pkgs/development/tools/gotestsum/default.nix]: :+1:  - TIL, thanks!

File: pkgs/development/tools/gotestsum/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    repo = "gotestsum";
+    rev = "v${version}";
+    sha256 = "1d4sbvk9wqzl3g3da8inqdkvd43rkwvmq969jlgl1k1agv5xjxqv";
+  };
+
+  modSha256 = "1dgs643pmcw68yc003zss52hbvsy6hxzwkrhr0qmsqkmzxryb3bn";
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/gotestyourself/gotestsum;
+    description = "A human friendly `go test` runner";
```
Comment (aanderse) [inline pkgs/development/tools/gotestsum/default.nix]: Please quote.

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -1822,6 +1822,12 @@
     githubId = 18535642;
     name = "Emily";
   };
+  endocrimes = {
```
Comment (aanderse) [inline maintainers/maintainer-list.nix]: Please include in a separate commit.

**Review by aanderse — CHANGES_REQUESTED** [review]
2 minor things. Looking good though. Thank you!

File: pkgs/development/tools/gotestsum/default.nix (hunk 3 above)
Comment (kalbasit) [reply pkgs/development/tools/gotestsum/default.nix]: I think you mean the homepage, right?

Comment (endocrimes) [reply pkgs/development/tools/gotestsum/default.nix]: Quoted the homepage now :+1:

File: maintainers/maintainer-list.nix (hunk 4 above)
Comment (endocrimes) [reply maintainers/maintainer-list.nix]: done :+1:

**Comment by aanderse** [comment]
@GrahamcOfBorg build gotestsum

**Review by aanderse — APPROVED** [review]
Thanks for contributing to nixpkgs and adding yourself to the list of maintainers... Welcome to the team! :tada:

**Pull request merged**